android/build.gradle

```diff
@@ -66,16 +66,16 @@ repositories {
 }
 
 dependencies {
-  def googlePlayServicesVisionVersion = safeExtGet('googlePlayServicesVisionVersion', safeExtGet('googlePlayServicesVersion', '17.0.2'))
-
   //noinspection GradleDynamicVersion
   implementation 'com.facebook.react:react-native:+'  // From node_modules
   implementation "com.google.zxing:core:3.3.3"
   implementation "com.drewnoakes:metadata-extractor:2.11.0"
-  generalImplementation "com.google.android.gms:play-services-vision:$googlePlayServicesVisionVersion"
   implementation "androidx.exifinterface:exifinterface:1.3.2"
   implementation "androidx.annotation:annotation:1.0.0"
   implementation "androidx.legacy:legacy-support-v4:1.0.0"
-  mlkitImplementation "com.google.firebase:firebase-ml-vision:${safeExtGet('firebase-ml-vision', '19.0.3')}"
-  mlkitImplementation "com.google.firebase:firebase-ml-vision-face-model:${safeExtGet('firebase-ml-vision-face-model', '17.0.2')}"
+  implementation "com.google.android.gms:play-services-mlkit-text-recognition:16.3.0"
+  generalImplementation "com.google.android.gms:play-services-mlkit-barcode-scanning:16.2.0"
+  generalImplementation "com.google.android.gms:play-services-mlkit-face-detection:16.2.0"
+  mlkitImplementation "com.google.mlkit:barcode-scanning:16.2.0"
+  mlkitImplementation "com.google.mlkit:face-detection:16.1.2"
```

Comment:
It is really unfortunate that they do not publish a "Bill of Materials" that handles all the disparate versioning, there appears to be no avoiding a different version for almost every part.
It might still be a good idea for a follow-on PR that allowed users to specify versions in gradle variables in an ext block like `googlePlayServicesVisionVersion` used to have, e.g. `playServicesMLKitFaceDetectionVersion` / `mlKitFaceDetectionVersion` etc etc